### ISSUE TITLE
TRACE security considerations (Trac #33)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#33]
```text
#!html
            <p>There is an HTTP-related security violation approach found/researched
            by White Hat Security:</p>
            
            <p>
                PR: <a href="http://www.whitehatsec.com/press_releases/WH-PR-20030120.txt">http://www.whitehatsec.com/press_releases/WH-PR-20030120.txt</a><br/>
                Details: <a href="http://www.betanews.com/whitehat/WH-WhitePaper_XST_ebook.pdf">http://www.betanews.com/whitehat/WH-WhitePaper_XST_ebook.pdf</a>
            </p>
            
            <p>I bet many of you have seen the related advisories/PR. For those who
            have not, here is the gist:</p>
            
            <p>Modern browsers usually do not allow scripts embedded in
            HTML to access cookies and authentication information
            exchanged between HTTP client and server. However, a
            script can get access to that info by sending a
            simple HTTP TRACE request to the originating (innocent)
            server. The user agent will auto-include current
            authentication info in such request. The server will echo all
            the authentication information back, for script to read and
            [mis]use. Apparently, sending an HTTP request is possible via
            many scripting methods like ActiveX. See the URL above for
            details.</p>
            
            <p>With numerous XSS (cross-site-scripting) vulnerabilities in user
            agents, this seems like a real and nasty problem. TRACE method support
            is optional per RFC 2616, but many popular servers support it. White
            Hat Security advises server administrators to disable support for
            TRACE.</p>
```

Migrated from https://trac.ietf.org/ticket/33
```json
{
    "status": "closed",
    "changetime": "2009-07-29T10:58:43",
    "_ts": "1248865123000000",
    "description": "{{{\n#!html\n            <p>There is an HTTP-related security violation approach found/researched\n            by White Hat Security:</p>\n            \n            <p>\n                PR: <a href=\"http://www.whitehatsec.com/press_releases/WH-PR-20030120.txt\">http://www.whitehatsec.com/press_releases/WH-PR-20030120.txt</a><br/>\n                Details: <a href=\"http://www.betanews.com/whitehat/WH-WhitePaper_XST_ebook.pdf\">http://www.betanews.com/whitehat/WH-WhitePaper_XST_ebook.pdf</a>\n            </p>\n            \n            <p>I bet many of you have seen the related advisories/PR. For those who\n            have not, here is the gist:</p>\n            \n            <p>Modern browsers usually do not allow scripts embedded in\n            HTML to access cookies and authentication information\n            exchanged between HTTP client and server. However, a\n            script can get access to that info by sending a\n            simple HTTP TRACE request to the originating (innocent)\n            server. The user agent will auto-include current\n            authentication info in such request. The server will echo all\n            the authentication information back, for script to read and\n            [mis]use. Apparently, sending an HTTP request is possible via\n            many scripting methods like ActiveX. See the URL above for\n            details.</p>\n            \n            <p>With numerous XSS (cross-site-scripting) vulnerabilities in user\n            agents, this seems like a real and nasty problem. TRACE method support\n            is optional per RFC 2616, but many popular servers support it. White\n            Hat Security advises server administrators to disable support for\n            TRACE.</p>\n}}}",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-20T05:06:11",
    "component": "p2-semantics",
    "summary": "TRACE security considerations",
    "priority": "normal",
    "keywords": "",
    "milestone": "unassigned",
    "owner": "",
    "type": "editorial",
    "severity": "Active WG Document"
}
```

**Comment by mnot** [comment]
Trac update at 20080104T06:16:31: 
* mnot@pobox.com changed component from "" to "semantics"
* mnot@pobox.com changed milestone from "" to "unassigned"
* mnot@pobox.com changed owner from "" to ""

**Comment by mnot** [comment]
Trac update at 20080104T06:30:34: mnot@pobox.com changed version from "" to "00-draft"

**Comment by mnot** [comment]
Trac update at 20080205T03:29:22: fielding@gbiv.com commented:

> That security advisory is a load of festering smelly bits.
> It depends on a known security issue on a known hole-filled
> browser and its related error-prone configuration option.
> The obvious solution (from a protocol perspective) is to
> avoid sending arbitrary private browser information in
> arbitrary methods executed by arbitrary javascript.  That has
> nothing to do with TRACE.
> 
> XSS should be discussed in the security considerations.

**Comment by mnot** [comment]
Trac update at 20081115T00:51:03: ylafon@w3.org commented:

> Proposal
> Add in the Security Considerations, in the "Transfer of Sensitive Information" section the following text:
> 
> <t>
>   Some methods, like TRACE (<xref target="method.trace" />) may expose
>   information sent in request headers in the response entity. Clients
>   &SHOULD; be careful with sensitive information, like Cookies, 
>   Authorization credentials and other headers that might be used to
>   collect data from the Client.
> </t>

**Comment by mnot** [comment]
Trac update at 20090729T10:46:07: 
* mnot@pobox.com changed priority from "" to "normal"
* mnot@pobox.com changed severity from "" to "Active WG Document"
* mnot@pobox.com changed type from "design" to "editorial"

**Comment by mnot** [comment]
Trac update at 20090729T10:58:43: 
* ylafon@w3.org commented:

> Fixed in [654]:
> 
> Resolve #33: Added TRACE security considerations (closes #33)
> 


* ylafon@w3.org changed resolution from "" to "fixed"
* ylafon@w3.org changed status from "new" to "closed"
